Formula/wasm-micro-runtime.rb

```diff
@@ -0,0 +1,41 @@
+class WasmMicroRuntime < Formula
+  desc "WebAssembly Micro Runtime (WAMR)"
+  homepage "https://github.com/bytecodealliance/wasm-micro-runtime"
+  url "https://github.com/bytecodealliance/wasm-micro-runtime/archive/refs/tags/WAMR-05-18-2022.tar.gz"
+  sha256 "c71f571f99c7028fd7595126ec25d6df4bc9ba62702bb88cbd5ea8a29e477f12"
+  license "Apache-2.0"
+  head "https://github.com/bytecodealliance/wasm-micro-runtime.git", branch: "main"
+
+  depends_on "cmake" => :build
+
+  def install
+    cmake_args = std_cmake_args + %w[
+      -DWAMR_BUILD_MULTI_MODULE=1
+      -DWAMR_BUILD_DUMP_CALL_STACK=1
+    ]
+
+    cmake_source = "product-mini/platforms/#{OS.mac? ? "darwin" : "linux"}"
+
+    system "cmake", "-S", cmake_source, "-B", "build", *cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    rm_r "build/"
+
+    system "cmake", "-S", ".", "-B", "build", *cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    rm_r "build/"
+  end
+
+  test do
+    resource "fib_wasm" do
+      url "https://github.com/wasm3/wasm3/raw/main/test/lang/fib.c.wasm"
+      sha256 "e6fafc5913921693101307569fc1159d4355998249ca8d42d540015433d25664"
+    end
```

Comment:
This should be outside the test block

Reply:
Doesn't' it always download the test even if I just simply install the formula? Or tests are run after that as well?

Reply:
Tests do not run or download unless you do `brew test <formula>`

Reply:
If it is outside the test function I get this:
```
brew install --build-from-source wasm-micro-runtime
Running `brew update --auto-update`...
==> Downloading https://github.com/wasm3/wasm3/raw/main/test/lang/fib.c.wasm
Already downloaded: /Users/nandork@cisco.com/Library/Caches/Homebrew/downloads/f3772ed663603103793e800290f735adf82286131255f4ef038ca33bc341c89d--fib.c.wasm
....
```

Reply:
That's only when building from source.

Reply:
So I can move it out from the test block?